### PR TITLE
feat(queue-events): add QueueEventsProducer for publishing custom events

### PR DESCRIPTION
**Pull request by roggervalf**
Documentation is missing
ref https://github.com/taskforcesh/bullmq/issues/2759

### CODE REVIEW

**Review by manast — COMMENTED** [review]
LGTM, however a proper documentation chapter will be required for this, maybe as a subchapter of global events.

File: src/classes/queue-events.ts [hunk 1]
```diff
@@ -307,7 +316,8 @@ export class QueueEvents extends QueueBase {
             this.emit(event, id);
           } else {
             this.emit(event as any, restArgs, id);
-            this.emit(`${event}:${restArgs.jobId}` as any, restArgs, id);
+            if (restArgs.jobId)
+              {this.emit(`${event}:${restArgs.jobId}` as any, restArgs, id);}
```
[the remark this answers is not in the document]
Comment (roggervalf) [reply src/classes/queue-events.ts]: yeah for some reason, lint didn't get applied

**Comment by vidjuheffex** [comment]
Very cool, subscribed to this one for when it drops. Did a similar setup to the "steps" pattern in the docs but without an event for the step changes, it's a bit clunky at parts.

thanks for this.

**Review by manast — APPROVED** [review]
LGTM

**Pull request merged**